src/Sylius/Bundle/CoreBundle/Checker/CatalogPromotionEligibilityChecker.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\CoreBundle\Checker;
+
+use Sylius\Bundle\PromotionBundle\Provider\EligibleCatalogPromotionProviderInterface;
+use Sylius\Component\Core\Model\CatalogPromotionInterface;
+
+final class CatalogPromotionEligibilityChecker implements CatalogPromotionEligibilityCheckerInterface
+{
+    public function __construct(private EligibleCatalogPromotionProviderInterface $eligibleCatalogPromotionProvider)
+    {
+    }
+
+    public function isCatalogPromotionEligible(CatalogPromotionInterface $promotion): bool
+    {
+        return $this->eligibleCatalogPromotionProvider->provide($promotion) !== null;
```

Comment:
You are providing here a catalog promotion to get a catalog promotion 🤔 And you are executing an additional query to DB for checking if the catalog promotion is eligible. I would be for refactoring it a bit by adding an additional method `check` or `verify` to `CriteriaInterface` to check a criterion in PHP and here use it to check all criteria.

Reply:
Ok, you are right 😃 I've added a new method to `CriteriaInterface` and checking criterion in PHP.